File: API/Todo/Todo.DTOs/Responses/TaskResponse.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Todo.Commons.Enums;

namespace Todo.DTOs.Responses
{
    public class TaskResponse
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime DueDate { get; set; }
        public bool IsCompleted { get; set; }
        public Tier Priority { get; set; }
        public DateTime? CompletedOn { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
    }
}

File: UI/src/interfaces/Responses/TaskResponse.ts
import type dayjs from "dayjs";
import type { Tier } from "../../commons";

export default interface TaskResponse {
  id?: string,
  title: string;
  description: string;
  dueDate: dayjs.Dayjs;
  isCompleted: boolean;
  priority: Tier;
  completedOn: dayjs.Dayjs;

  createdOn?: dayjs.Dayjs;
  // createdBy?: string;
  modifiedOn?: dayjs.Dayjs;
}